File: rzops-web/src/lib/types/certificate.ts
export interface CertificateResponse {
  id: string;
  name: string;
  provider_id: string | null;
  lease_start_date: string | null;
  lease_end_date: string | null;
  certificate_type: string | null;
  status: string;
  remarks: string | null;
  created_at: string;
  updated_at: string;
}

export interface CertificateListResponse {
  data: CertificateResponse[];
  count: number;
}

export interface CreateCertificateRequest {
  name: string;
  provider_id?: string;
  lease_start_date?: string;
  lease_end_date?: string;
  certificate_type?: string;
  status?: string;
  remarks?: string;
}

export interface UpdateCertificateRequest extends Partial<CreateCertificateRequest> {}

export interface ListCertificatesQuery {
  q?: string;
  status?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/certificate.rs
use chrono::{DateTime, NaiveDate, Utc};
use uuid::Uuid;


/// Certificate entity — maps to cmdb_certificate.
#[derive(Debug, Clone)]
pub struct Certificate {
    pub id: Uuid,
    pub name: String,
    pub provider_id: Option<Uuid>,
    pub lease_start_date: Option<NaiveDate>,
    pub lease_end_date: Option<NaiveDate>,
    pub certificate_type: Option<String>,
    pub status: String,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
